Unosquare.CodeChallenge/WarehouseTestingUnit/WarehouseServices/WarehouseServices_unitTest.cs

```diff
@@ -0,0 +1,145 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Tests;
+using WarehouseCoreLib.DataAccess;
+using WarehouseModels.Models;
+using WarehouseServices.Services;
+using Assert = NUnit.Framework.Assert;
+using Theory = NUnit.Framework.TheoryAttribute;
+
+namespace WarehouseTestingUnit.WarehouseServices
+{
+    public class WarehouseServices_unitTest
+    {
+
+        [Fact]
+        public void Test_getAllCompanies_getAllCompaniesItems()
+        {
+            CompaniesServices service = new CompaniesServices(WarehouseHelper.createDBContext());
+            List<Company> companies = service.GetAllCompanies().ToList();
+
+            Assert.IsNotEmpty(companies);
+            Assert.Greater(companies.Count, 0);
+        }
+
+        [Fact]
+        public async void Test_getAllCompaniesAsync_getAllCompaniesItems()
+        {
+            CompaniesServices service = new CompaniesServices(WarehouseHelper.createDBContext());
+            List<Company> companies = await  service.GetAllCompaniesAsync();
+
+            Assert.IsNotEmpty(companies);
+            Assert.Greater(companies.Count, 0);
+        }
+
+        [Fact]
+        public void Test_GetCompany_GetItemCompanyById()
+        {
+            CompaniesServices service = new CompaniesServices(WarehouseHelper.createDBContext());
+            Company company = service.GetCompany(1);
+
+            Assert.AreEqual(1, company.Id);
+
+        }
+
+        [Fact]
+        public async void Test_GetCompanyAsync_GetItemCompanyByID()
+        {
+            CompaniesServices service = new CompaniesServices(WarehouseHelper.createDBContext());
+            Company company = await service.GetCompanyAsync(1);
+
+            Assert.AreEqual(1, company.Id);
+        }
+
+        [Fact]
+        public void Test_AddCompany_addItemToDb()
+        {
+            Company newCompany = new Company()
+            {
+                Name = "new company test unit"
+            };
+
+            CompaniesServices service = new CompaniesServices(WarehouseHelper.createDBContext());
+            service.AddCompany(newCompany);
+            Assert.True(true);
+        }
+
+
+        [Fact]
+        public async void Test_AddCompanyAsync_addItemToDb()
```

Comment:
Test_ is redundant, we can discuss two different ways to name unit tests in a meeting later, just make sure to remind me.

Reply:
got it...I'll will remind you